firewood/src/merkle.rs

```diff
@@ -1491,28 +1491,22 @@ mod tests {
         assert_eq!(encoded, new_node_encoded);
     }
 
-    #[test_case(leaf(Vec::new(), Vec::new()) ; "empty leaf encoding")]
-    #[test_case(leaf(vec![1, 2, 3], vec![4, 5]) ; "leaf encoding")]
-    #[test_case(branch(Some(b"value".to_vec()), vec![1, 2, 3].into()) ; "branch with chd")]
-    #[test_case(branch(Some(b"value".to_vec()), None); "branch without chd")]
-    #[test_case(branch(None, None); "branch without value and chd")]
-    fn node_encode_decode(node: Node) {
-        let merkle = create_test_merkle();
-        let node_ref = merkle.put_node(node.clone()).unwrap();
-
-        let encoded = merkle.encode(&node_ref).unwrap();
-        let new_node = Node::from(merkle.decode(encoded.as_ref()).unwrap());
-
-        assert_eq!(node, new_node);
-    }
-
-    #[test_case(leaf(Vec::new(), Vec::new()) ; "empty leaf encoding")]
-    #[test_case(leaf(vec![1, 2, 3], vec![4, 5]) ; "leaf encoding")]
-    #[test_case(branch(Some(b"value".to_vec()), vec![1, 2, 3].into()) ; "branch with chd")]
-    #[test_case(branch(Some(b"value".to_vec()), Some(Vec::new())); "branch with empty chd")]
-    #[test_case(branch(Some(Vec::new()), vec![1, 2, 3].into()); "branch with empty value")]
-    fn node_encode_decode_plain(node: Node) {
-        let merkle = create_generic_test_merkle::<PlainCodec>();
+    #[test_case(Bincode::new(), leaf(Vec::new(), Vec::new()) ; "empty leaf encoding with Bincode")]
```

Comment:
nit: this might work better with the [test-matrix](https://docs.rs/test-case/latest/test_case/#test-matrix) option, which seems to have a way of creating a cartesian product of possible tests.

I didn't test that it would work in this use case, but it might simplify this code and ensure that if we add new ones we don't miss both test cases.

Reply:
yeah, thought about that, although unless https://github.com/ava-labs/firewood/issues/433 is addressed, some test cases cannot be merged.

Reply:
woooow... `test-matrix` looks really nice and should definitely be used where we can.